inference-engine/src/vpu/graph_transformer/src/stages/gather_nd.cpp

```diff
@@ -0,0 +1,105 @@
+
+namespace {
+
+class GatherNDStage final : public StageNode {
+public:
+    using StageNode::StageNode;
+
+protected:
+    StagePtr cloneImpl() const override { return std::make_shared<GatherNDStage>(*this); }
+
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder> &orderInfo) override {
+        const auto input1 = inputEdge(0)->input();
+        const auto input2 = inputEdge(1)->input();
+        const auto output = outputEdge(0)->output();
+
+        orderInfo.setInput(inputEdge(0), DimsOrder::fromNumDims(input1->desc().numDims()));
+        orderInfo.setInput(inputEdge(1), DimsOrder::fromNumDims(input2->desc().numDims()));
+        orderInfo.setOutput(outputEdge(0), DimsOrder::fromNumDims(output->desc().numDims()));
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement> &stridesInfo) override {
+        for (const auto &inEdge : inputEdges()) {
+            stridesInfo.setInput(inEdge, StridesRequirement::compact());
+        }
+        stridesInfo.setOutput(outputEdge(0), StridesRequirement::compact());
+    }
+
+    void finalizeDataLayoutImpl() override {}
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport> &batchInfo) override {}
+
+    StageSHAVEsRequirements getSHAVEsRequirementsImpl() const override {
+        return StageSHAVEsRequirements::NotNeeded;
+    }
+
+    void initialCheckImpl() const override {
+        VPU_THROW_UNLESS(numInputs() == 2, "Number of inputs must be 2");
+        VPU_THROW_UNLESS(numOutputs() == 1, "Number of outputs must be 1");
+        VPU_THROW_UNLESS(inputs()[0]->desc().type() == outputs()[0]->desc().type(),
+                         "First input and output must have the same DataType");
+
+        assertInputsOutputsTypes(
+            this,
+            {{DataType::U8, DataType::FP16, DataType::S32}, {DataType::S32}},
+            {{DataType::U8, DataType::FP16, DataType::S32}});
+    }
+
+    void serializeParamsImpl(BlobSerializer &serializer) const override {
+        const auto batchDims = attrs().get<int32_t>("batch_dims");
+        serializer.append(batchDims);
+    }
+
+    void serializeDataImpl(BlobSerializer &serializer) const override {
+        auto input0 = inputEdge(0)->input();
+        auto input1 = inputEdge(1)->input();
+        auto output = outputEdge(0)->output();
+
+        input0->serializeBuffer(serializer);
+        output->serializeBuffer(serializer);
+        input1->serializeBuffer(serializer);
+    }
+};
+
+} // namespace
+
+void FrontEnd::parseGatherND(
+    const Model &model,
+    const ie::CNNLayerPtr &layer,
+    const DataVector &inputs,
+    const DataVector &outputs) const {
+    VPU_THROW_UNLESS(
+        inputs.size() == 2,
+        "GatherND parsing failed, expected number of inputs: 2, but {} provided. Layer name: {}",
+        inputs.size(),
+        layer->name);
+    VPU_THROW_UNLESS(
+        outputs.size() == 1,
+        "GatherND parsing failed, expected number of outputs: 1, but {} provided.  Layer name: {}",
```

Comment:
Excess whitespace. Also, provide an actual number of inputs/outputs, please

Reply:
Is there a style guide or `.clang-format` file I could use?

Reply:
> Also, provide an actual number of inputs/outputs, please

I'll add such a rule into the Style guide and send it to you later